### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by mrcnski**
Some minor, unrelated fixes. I'll add a bit more detail in comments.

### CODE REVIEW

File: node/subsystem-util/src/nesting_sender.rs [hunk 1]
```diff
@@ -196,10 +196,9 @@ where
 
 // Helper traits and implementations:
 
-impl<M, M1> Clone for NestingSender<M, M1>
+impl<M> Clone for NestingSender<M, M>
```
Comment (mrcnski) [inline node/subsystem-util/src/nesting_sender.rs:199]: `M, M1` was a relic from an old version of the file. I changed it to `M, M` to be consistent with the constructor.

File: node/core/pvf/src/execute/worker.rs [hunk 2]
```diff
@@ -142,11 +142,8 @@ pub async fn start_work(
 				target: LOG_TARGET,
 				worker_pid = %pid,
 				validation_code_hash = ?artifact.id.code_hash,
-				"execution worker exceeded allotted time for execution",
+				"execution worker exceeded lenient timeout for execution, child worker likely stalled",
 			);
-			// TODO: This case is not really a hard timeout as the timeout here in the host is
-			// lenient. Should fix this as part of
-			// https://github.com/paritytech/polkadot/issues/3754.
 			Response::TimedOut
```
Comment (mrcnski) [inline node/core/pvf/src/execute/worker.rs:147]: Just leaving this error as-is. If we hit this case we should just treat it the same as any other timeout, i.e. kill the worker. The log above should be good enough to inform users what happened.

I had some thoughts at the time about refactoring the execution errors, like we did for preparation. However, looking at the errors here, there don't seem to be any easy wins for simplicity or semantic correctness.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2246814

**Review by tdimitrov — APPROVED** [review]

**Review by s0me0ne-unkn0wn — APPROVED** [review]

**Comment by mrcnski** [comment]
bot merge

**Pull request merged**